server/src/controllers/course.ts

```diff
@@ -40,3 +53,146 @@ export async function addCourse(req: Request, res: Response): Promise<void> {
     });
   }
 }
+
+export async function getCourse(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: number = Number(req.params.courseId);
+    const course: courseService.CourseWithTranslationAndInstance = await courseService.findCourseById(courseId);
+  
+    const courseData: CourseData = {
+      id: course.id,
+      courseCode: course.courseCode,
+      minCredits: course.minCredits,
+      maxCredits: course.maxCredits,
+      department: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      name: {
+        en: '',
+        fi: '',
+        sv: ''
+      },
+      evaluationInformation: {
+        en: '',
+        fi: '',
+        sv: ''
+      }
+    };
+  
+    course.CourseTranslations.forEach((translation: CourseTranslation) => {
+      switch (translation.language) {
+      case Language.English:
+        courseData.department.en = translation.department;
+        courseData.name.en = translation.courseName;
+        break;
+      case Language.Finnish:
+        courseData.department.fi = translation.department;
+        courseData.name.fi = translation.courseName;
+        break;
+      case Language.Swedish:
+        courseData.department.sv = translation.department;
+        courseData.name.sv = translation.courseName;
+        break;
+      }
+    });
+  
+    res.status(200);
+    return res.send({
+      success: true,
+      course: courseData
+    });
+  } catch (error: unknown) {
+    console.log(error);
+  
+    if (error instanceof Error) {
+      res.status(404);
+      return res.send({
+        success: false,
+        error: error.message
+      });
+    }
+    res.status(500);
+    return res.send({
+      success: false,
+      error: 'Internal Server Error'
+    });
+  }
+}
+      
+export async function getInstance(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: number = Number(req.params.courseId);
```

Comment:
What if this produces a `NaN`?

Reply:
yup added https://github.com/aalto-grades/base-repository/pull/112/commits/58b3eaa39ed0167afab1113fd9bfd0d68ae98679